README
Acme-MetaSyntactic-soviet

This module is a theme for Acme::MetaSyntactic, giving the NATO
codenames for various Soviet equipments.

INSTALLATION

To install this module, use the CPAN shell or cpanminus.

Or you can use the manual way by running the following commands:

        perl Makefile.PL
        make
        make test
        make install

There is also a Build.PL file, but since Module::Build is no longer recommended,
this installation procedure has not been tested for this version.

SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the
perldoc command.

    perldoc Acme::MetaSyntactic::soviet

You can also look for information at:

    meta::CPAN
        https://metacpan.org/pod/Acme::MetaSyntactic::soviet

Please report problems with Github's issues tracker:
        https://github.com/jforget/Acme-MetaSyntactic-soviet/issues


COPYRIGHT AND LICENCE

Copyright (C) 2008, 2012, 2016, 2021 Jean Forget

This program is free software; you can redistribute it and you can modify it
under the same terms as Perl itself.



Changes
-*- encoding: utf-8; indent-tabs-mode: nil -*-

Revision history for Acme-MetaSyntactic-soviet

0.07    20xx-xx-xx, the xxx version
        Seventh version, released on the xxth anniversary of ...
        - xxx

0.06    2021-05-10, the Fresco version
        Sixth version, released on the 49th anniversary of the dogfight between a Fresco piloted by
        ace Colonel Tomb and a F-4 Phantom piloted by soon-to-be-aces Cunningham and Driscoll
        - new bugtracker: Github instead of rt.cpan.org
        - add data from magazine "Science & Vie" (special issue Aviation 1977), from the novel
        "Helicopter Aces", from the books "Bullseye one Reactor", "Air Warfare in the Missile Age",
        "Aircraft Armament"

0.05    2016-10-21, the Styx version
        Fifth version, released  on the 49th anniversary of the sinking of the Israeli destroyer Eilat by Styx missiles fired from Egyptian patrol boats.
        - playing the CPANTS game: adding "strict" and "warnings" as prerequisite.
        - add data from Dunnigan's "How to Make War" and from Clancy's "Red Storm Rising".

0.04    2016-05-01, the Guideline version
        Fourth version, released on the 56th anniversary of the downing of Gary Power's U-2 by a SA-2 Guideline.
        - fix some issues (duplicate entries)
        - add data from "Les Flottes de Combat 1968", SPI-TSR's "Air War", GTG's "Rolling Thunder",
        "The Hunt for Red October" and "Fox Two".

0.03    2012-09-06, the Foxbat version
        Third version, released on the 36th anniversary of Victor Belenko's flight from USSR to Japan.
        - Compatibility with Acme::MetaSyntactic::MultiList version 1.
        - Loading data from POD.
        - Adding submarines, which may be redundant with A::MS::alphabet (nato category),
        A::MS::phonetic (x-nato category) and A::MS::roman, but has a few more entries.

0.02    2012-01-02, the Fishbed version
        Second version, released on the 45th anniversary of Operation Bolo.
        - Various typos and sorting errors fixed.
        - Duplicate entries removed.
        - POD encoding clearly specified, so the website will correctly print FDC's author name.

0.01    2008-10-23, the Sandal version
        First version, released on the 46th anniversary of the blockade against Cuba



Build.PL
#
#     Configuration script for Acme::MetaSyntactic::soviet
#     Copyright (C) 2008, 2012, 2016, 2021 Jean Forget
#
#     This program is distributed under the same terms as Perl 5.16.3:
#     GNU Public License version 1 or later and Perl Artistic License
#
#     You can find the text of the licenses in the F<LICENSE> file or at
#     L<http://www.perlfoundation.org/artistic_license_1_0>
#     and L<http://www.gnu.org/licenses/gpl-1.0.html>.
#
#     Here is the summary of GPL:
#
#     This program is free software; you can redistribute it and/or modify
#     it under the terms of the GNU General Public License as published by
#     the Free Software Foundation; either version 1, or (at your option)
#     any later version.
#
#     This program is distributed in the hope that it will be useful,
#     but WITHOUT ANY WARRANTY; without even the implied warranty of
#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#     GNU General Public License for more details.
#
#     You should have received a copy of the GNU General Public License
#     along with this program; if not, write to the Free Software Foundation,
#     Inc., <http://www.fsf.org/>.
#
use strict;
use warnings;
use Module::Build;

my $builder = Module::Build->new(
    module_name         => 'Acme::MetaSyntactic::soviet',
    license             => 'perl',
    dist_author         => 'Jean Forget <JFORGET@cpan.org>',
    dist_version_from   => 'lib/Acme/MetaSyntactic/soviet.pm',
    dist_abstract       => 'Metasyntactic theme with NATO codenames for Soviet-designed equipment',
    configure_requires => {
        'Module::Build' => 0.33,
    },
    build_requires => {
        'Acme::MetaSyntactic::MultiList' => 1,
        'Test::More' => 0,
    },
    add_to_cleanup      => [ 'Acme-MetaSyntactic-soviet-*' ],
    create_makefile_pl => 'traditional',
);

$builder->create_build_script();


Makefile.PL
#
#     Configuration script for Acme::MetaSyntactic::soviet
#     Copyright (C) 2008, 2012, 2016, 2021 Jean Forget
#
#     This program is distributed under the same terms as Perl 5.16.3:
#     GNU Public License version 1 or later and Perl Artistic License
#
#     You can find the text of the licenses in the F<LICENSE> file or at
#     L<https://dev.perl.org/licenses/artistic.html>
#     and L<https://www.gnu.org/licenses/gpl-1.0.html>.
#
#     Here is the summary of GPL:
#
#     This program is free software; you can redistribute it and/or modify
#     it under the terms of the GNU General Public License as published by
#     the Free Software Foundation; either version 1, or (at your option)
#     any later version.
#
#     This program is distributed in the hope that it will be useful,
#     but WITHOUT ANY WARRANTY; without even the implied warranty of
#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#     GNU General Public License for more details.
#
#     You should have received a copy of the GNU General Public License
#     along with this program; if not, contact the Free Software Foundation,
#     Inc., <https://www.fsf.org/>.
#
# Note: this file was auto-generated by Module::Build::Compat version 0.4001
use ExtUtils::MakeMaker;
WriteMakefile
(
    'ABSTRACT'       => 'Metasyntactic theme with NATO codenames for Soviet-designed equipment',
    'PL_FILES'       => {},
    'INSTALLDIRS'    => 'site',
    'NAME'           => 'Acme::MetaSyntactic::soviet',
    'EXE_FILES'      => [],
    'VERSION_FROM'   => 'lib/Acme/MetaSyntactic/soviet.pm',
    'PREREQ_PM' => {
                     'Test::More'                     => 0,
                     'Acme::MetaSyntactic::MultiList' => 1,
                     'strict'                         => 0,
                     'warnings'                       => 0,
                   },
    'AUTHOR'         => [ 'Jean Forget <JFORGET@cpan.org>' ],
    'LICENSE'        => 'perl',
    # The oldest Perl to check A::MS::s 0.03 is 5.6.2. Therefore, I guess A::MS::s 0.04 and next will work in 5.6.2 too.
    MIN_PERL_VERSION => '5.6.2',
    META_MERGE       => {
       dynamic_config => 0,
       prereqs => {
         build => {
           requires => {
             'strict'                         => 0,
             'warnings'                       => 0,
           },
         },
         runtime => {
           requires => {
             'Acme::MetaSyntactic::MultiList' => 1,
             'perl'                           => '5.6.2',
             'strict'                         => 0,
             'warnings'                       => 0,
           },
         },
         test => {
           requires => {
             'Test::More'                     => '0',
             'strict'                         => 0,
             'warnings'                       => 0,
           },
         },
       },
       provides       => {
           'Acme::MetaSyntactic::soviet' => {
                file => 'lib/Acme/MetaSyntactic/soviet.pm',
                version => '0.07',
           },
       },
       'meta-spec' => { version => 2 },
       resources => {
           repository => {
               type => 'git',
               url  => 'https://github.com/jforget/Acme-MetaSyntactic-soviet.git',
               web  => 'https://github.com/jforget/Acme-MetaSyntactic-soviet',
           },
           bugtracker => {
               web  => 'https://github.com/jforget/Acme-MetaSyntactic-soviet/issues',
           },
       },
    },
)
;
